dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/AppClassLoader.java

```diff
@@ -702,13 +701,28 @@ final ByteResourceInformation findClassBytes(String className, String resourceNa
      * Find a class on this loader's class path or delegate to the parent class
      * loader.
      */
+    @FFDCIgnore(ClassNotFoundException.class)
     protected Class<?> findOrDelegateLoadClass(String name) throws ClassNotFoundException {
-        // The resolve parameter is a legacy parameter that is effectively
-        // never used as of JDK 1.1 (see footnote 1 of section 5.3.2 of the 2nd
-        // edition of the JVM specification).  The only caller of
-        // loadClass(String, boolean) is java.lang.ClassLoader.loadClass(String),
-        // and that method always passes false, so we ignore the parameter.
-        return super.loadClass(name, false);
+        Class<?> result = findLoadedClass(name);
+        if (result != null) {
+            return result;
+        }
+        if (parent instanceof NoClassNotFoundLoader) {
+            result = ((NoClassNotFoundLoader) parent).loadClassNoException(name);
+        } else {
+            try {
+                if (parent == null) {
+                    return super.loadClass(name, false);
+                }
+                result = parent.loadClass(name);
```

Comment:
Should we pass false to this loadClass as well?

Reply:
I find this annoying about Java, but a subclass can only call protected methods on the super instance or other types that are in the same package (or self).